src/main/scala/org/broadinstitute/dsde/workbench/leonardo/auth/sam/SamAuthProvider.scala

```diff
@@ -174,6 +174,8 @@ class SamAuthProvider(val config: Config, serviceAccountProvider: ServiceAccount
       Future {
         blocking(samClient.deleteNotebookClusterResource(creatorEmail, googleProject, clusterName))
       }.recover {
+        // treat 404s from Sam as the cluster already being deleted
+        case e: ApiException if e.getCode == StatusCodes.NotFound.intValue => ()
```

Comment:
I observed `ClusterMonitorActor` going into an infinite loop because of this. If the resource was already deleted in Sam, then `notifyClusterDeleted()` would throw an exception, causing the actor to crash and be recreated. This scenario _shouldn't_ happen anymore due to the changes in `LeonardoService`, but still it makes sense to recover 404s here.